File: python/two_sum.py
def two_sum(numbers, target):
	index_map = {}
	
	for i in range(len(numbers)):
		number = numbers[i]
		pair = target - number
		if pair in index_map:
			return (index_map[pair], i)
		index_map[number] = i
	
	return -1

File: typescript/twoSum.ts
// Given an array of integers, return the indices of
// the two numbers that add up to a given target.

function twoSum(nums: number[], target: number): number[] | null {
  const seen: { [k: number]: number } = {};

  for (let i = 0; i < nums.length; i++) {
    const diff = target - nums[i];

    if (diff in seen) return [seen[diff], i];
    seen[nums[i]] = i;
  }

  return null;
}

console.log(twoSum([1, 3, 7, 9, 2], 11));
